esp8266-fastled-webserver/data/index.htm

```diff
@@ -7,12 +7,45 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <title>ESP8266 + FastLED</title>
 
-  <!-- request CSS from internet CDN -->
-  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
-  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-minicolors/2.3.6/jquery.minicolors.min.css" integrity="sha512-BVeRnUOL0G7d4gXmj+0VxpoiQuEibKQtlkclADKvCdNrESs0LA6+H8s1lU455VqWFtHBfF/pKDGw/CMat2hqOg==" crossorigin="anonymous" referrerpolicy="no-referrer" />
-  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.7.2/font/bootstrap-icons.min.css">
+  <!-- Fallback to load resources from FileSystem if not available online -->
+  <script>
+    function CssJsFallback(el) {
+      if (el instanceof HTMLScriptElement || el instanceof HTMLLinkElement) {
+        let Url = el.getAttribute('data-fallback');
+        if (Url) {
+          if(el instanceof HTMLScriptElement)
+          {
+            document.write('<script src="'+Url+'">\x3C/script>');
+          }
+          else
+          {
+            el.setAttribute('href', Url);  
+          }
+        }
+        // No integrity check for local resource
+        let integrity = el.getAttribute('integrity');
+        if(integrity) {
+          el.removeAttribute('integrity');
+        }
+      }
+    }
+  </script>
```

Comment:
This appears to modify the attributes of the element that fails to load via use of the `onerror()` callback, and for this question, I will presume that modification of the element will force a web browser to attempt to reload the script.

Question:
If there is a problem loading the file from the original URI, what happens if the attempt to load from the local file system also fails?

Will this go into an infinite loop?

If so, would it be appropriate to change or clear the `onerror` attribute here?

Reply:
On the second attempt (the fallback) it will try once more but then the fallback would be equal to the current element.
If it works, it looksmlike this:
![image](https://user-images.githubusercontent.com/1083336/155984762-b41fe33a-fadb-4e3d-a39a-dfeef183e019.png)

If both are wrong it just stops after the second one (maybe as there is nothing to be replaced being the same path):
![image](https://user-images.githubusercontent.com/1083336/155985169-7d27217c-30fd-4c4e-9e05-c8559fdb0e8e.png)